pkg/verifier/package_verifier.go

```diff
@@ -0,0 +1,59 @@
+package verifier
+
+import (
+	"context"
+	"time"
+
+	"github.com/openshift/osd-network-verifier/pkg/output"
+)
+
+// VerifierService defines the behaviors necessary to run verifier completely. Any clients that use that fullfills this interface
+// will be able to run all verifier test
+type verifierService interface {
+
+	// ByoVPCValidator validates the configuration given by the customer
+	ByoVPCValidator(bvvi ByoVPCValidatorInput) error
+
+	// ValidateEgress validates that all required targets are reachable from the vpcsubnet
+	// target URLs: https://docs.openshift.com/rosa/rosa_getting_started/rosa-aws-prereqs.html#osd-aws-privatelink-firewall-prerequisites
+	// Expected return value is *output.Output that's storing failures, exceptions and errors
+	ValidateEgress(vei ValidateEgressInput) *output.Output
+
+	// VerifyDns verifies that a given VPC meets the DNS requirements specified in:
+	// https://docs.openshift.com/container-platform/4.10/installing/installing_aws/installing-aws-vpc.html
+	// Expected return value is *output.Output that's storing failures, exceptions and errors
+	VerifyDns(vdi VerifyDnsInput) *output.Output
+}
+
+type ByoVPCValidatorInput struct {
+	Ctx context.Context
+}
+
+// ByoVPCValidator pass in a GCP or AWS client that know how to fufill above interface
+func ByoVPCValidator(vs verifierService, Bvvi ByoVPCValidatorInput) error {
+	return vs.ByoVPCValidator(Bvvi)
+}
+
+type ValidateEgressInput struct {
+	Timeout                             time.Duration
+	Ctx                                 context.Context
+	VpcSubnetID, CloudImageID, KmsKeyID string
+	//AWS
+	InstanceType string
```

Comment:
why are those specific to AWS? Wouldn't we set values for `InstanceType` or `Tags` on GCP verification tests?

Reply:
I added this comment because its currently whats used for AWS. I am not sure if the instancetypes or tags will be the same on gcp , but since there are no gcp implementations I guess it doesn't matter right now anyways : )